packages/x-charts/src/ChartsLegend/ChartsLegend.tsx

```diff
@@ -1,97 +1,137 @@
 'use client';
 import * as React from 'react';
+import { styled, SxProps, Theme } from '@mui/material/styles';
 import PropTypes from 'prop-types';
-import useSlotProps from '@mui/utils/useSlotProps';
-import composeClasses from '@mui/utils/composeClasses';
-import { DefaultizedProps } from '@mui/x-internals/types';
-import { useThemeProps, useTheme, Theme } from '@mui/material/styles';
-import { getSeriesToDisplay } from './utils';
-import { getLegendUtilityClass } from './chartsLegendClasses';
-import { DefaultChartsLegend, LegendRendererProps } from './DefaultChartsLegend';
-import { useSeries } from '../hooks/useSeries';
-import { LegendPlacement } from './legend.types';
-
-export type ChartsLegendPropsBase = Omit<
-  LegendRendererProps,
-  keyof LegendPlacement | 'series' | 'seriesToDisplay' | 'drawingArea'
-> &
-  LegendPlacement;
-
-export interface ChartsLegendSlots {
+import clsx from 'clsx';
+import { useLegend } from '../hooks/useLegend';
+import type { Direction } from './direction';
+import { SeriesLegendItemContext } from './legendContext.types';
+import { ChartsLabelMark } from '../ChartsLabel/ChartsLabelMark';
+import { seriesContextBuilder } from './onClickContextBuilder';
+import { legendClasses, useUtilityClasses, type ChartsLegendClasses } from './chartsLegendClasses';
+import { consumeSlots } from '../internals/consumeSlots';
+import { ChartsLabel } from '../ChartsLabel';
+
+export interface ChartsLegendProps {
   /**
-   * Custom rendering of the legend.
-   * @default DefaultChartsLegend
+   * Callback fired when a legend item is clicked.
+   * @param {React.MouseEvent<HTMLButtonElement, MouseEvent>} event The click event.
+   * @param {SeriesLegendItemContext} legendItem The legend item data.
+   * @param {number} index The index of the clicked legend item.
    */
-  legend?: React.JSXElementConstructor<LegendRendererProps>;
-}
-
-export interface ChartsLegendSlotProps {
-  legend?: Partial<LegendRendererProps>;
-}
-
-export interface ChartsLegendProps extends ChartsLegendPropsBase {
+  onItemClick?: (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    legendItem: SeriesLegendItemContext,
+    index: number,
+  ) => void;
   /**
-   * Overridable component slots.
-   * @default {}
+   * The direction of the legend layout.
+   * The default depends on the chart.
    */
-  slots?: ChartsLegendSlots;
+  direction?: Direction;
```

Comment:
We might want to change `direction` into `layout` our `orientation` as direction is not really correct here